### PR TITLE
JESI-3369

### PR DESCRIPTION
**Pull request by AndreTheHunter**
[JESI-3369](https://jesims.atlassian.net/browse/JESI-3369)

### CODE REVIEW

File: project.sh [hunk 1]
```diff
@@ -562,11 +562,17 @@ local-clean(){
...
 	allow-snapshots
 	local cmd
+	#TODO make args order independent
 	case $1 in
 		-r|--refresh|--watch)
 			cmd='--watch'
 			shift;;
 	esac
+	case $1 in
+		-ff|--fail-fast)
```
Comment (AndreTheHunter) [inline project.sh]: Friendly fire

Comment (axrs) [reply project.sh]: What if you pass the order `-ff -r`?

Comment (AndreTheHunter) [reply project.sh]: I won't pick up the `-r`. Thus the TODO.
Suggestions welcome

Comment (axrs) [reply project.sh]:
```
-test-clj () {
        allow-snapshots
	local cmd
	local remaining
	while [[ -n "$1" ]];do
		case "$1" in
			-r|--refresh|--watch)
				cmd="$cmd --watch";;
			-ff|--fail-fast)
				cmd="$cmd --fail-fast";;
			*)
				remaining="$remaining $1";;
		esac
		shift
	done
	cmd=$(echo "$cmd" | xargs)
	remaining=$(echo "$remaining" | xargs)
	if [ -n "$remaining" ];then
		cmd="$cmd --focus $remaining"
	fi
	...
}
```

File: project.sh [hunk 2]
```diff
@@ -559,22 +559,34 @@ local-clean(){
...
-		cmd="$cmd --focus $*"
+	local remaining
+	while [ -n "$1" ];do
+		case "$1" in
+			-r|--refresh|--watch)
+				cmd="$cmd --watch";;
+			-ff|--fail-fast)
+				cmd="$cmd --fail-fast";;
+			*)
+				remaining="$remaining $*";;
```
Comment (axrs) [inline project.sh]: Not sure you want to use $* here.

`-test-clj -ff -r io.jesi.blah io.jesi.not-blah` becomes
`--fail-fast --watch --focus io.jesi.blah io.jesi.not-blah io.jesi.not-blah`

the loop will collate remaining args as it goes
```suggestion
				remaining="$remaining $1";;
```

**Review by axrs — CHANGES_REQUESTED** [review]

**Review by axrs — APPROVED** [review]

**Pull request merged**